src/main/java/org/apache/commons/imaging/formats/tiff/TiffRasterDataInt.java

```diff
@@ -0,0 +1,184 @@
+    @Override
+    public void setValue(final int x, final int y, final float value) {
+        int index = checkCoordinates(x, y);
+        data[index] = (int)value;
+    }
+
+    /**
+     * Gets the value stored at the specified raster coordinates.
+     *
+     * @param x integer coordinate in the columnar direction
+     * @param y integer coordinate in the row direction
+     * @return the value stored at the specified location; potentially a
+     * Float&#46;NaN.
+     */
+    @Override
+    public float getValue(final int x, final int y) {
+        int index = checkCoordinates(x, y);
+        return data[index];
+    }
+
+
+    /**
+     * Sets the value stored at the specified raster coordinates.
+     *
+     * @param x integer coordinate in the columnar direction
+     * @param y integer coordinate in the row direction
+     * @param value the value to be stored at the specified location
+     */
+    @Override
+    public void setIntValue(final int x, final int y, final int value) {
+        int index = checkCoordinates(x, y);
+        data[index] = value;
+    }
+
+    /**
+     * Gets the value stored at the specified raster coordinates.
+     *
+     * @param x integer coordinate in the columnar direction
+     * @param y integer coordinate in the row direction
+     * @return the value stored at the specified location
+     */
+    @Override
+    public int getIntValue(final int x, final int y) {
+        int index = checkCoordinates(x, y);
+        return data[index];
+    }
+
+
+
+    /**
+     * Tabulates simple statistics for the raster and returns an instance
+     * containing general metadata.
+     *
+     * @return a valid instance containing a safe copy of the current simple
+   * statistics for the raster.     */
+    @Override
+    public TiffRasterStatistics getSimpleStatistics() {
+        return new TiffRasterStatistics(this, Float.NaN);
+    }
+
+    /**
+     * Tabulates simple statistics for the raster excluding the specified value
+     * and returns an instance containing general metadata.
+     *
+     * @param valueToExclude exclude samples with this specified value.
+     * @return a valid instance.
+     */
+    @Override
+    public TiffRasterStatistics getSimpleStatistics(final float valueToExclude) {
+        return new TiffRasterStatistics(this, valueToExclude);
+    }
+
+
+    /**
+     * Returns a reference to the data array stored in this instance. Note that
+     * the array returned is <strong>not</strong> a safe copy and that modifying
+     * it directly affects the content of the instance. While this design approach
+     * carries some risk in terms of data security, it was chosen for reasons of
+     * performance and memory conservation. TIFF images that contain
+     * floating-point data are often quite large. Sizes of 100 million raster
+     * cells are common. Making a redundant copy of such a large in-memory object
+     * might exceed the resources available to a Java application.
```

Comment:
Ditto previous comments above moving this important information to the top of the class and mentioning it here (or not).

Reply:
done